pkg/cloudprovider/azure/block_store.go

```diff
@@ -233,11 +250,19 @@ func (b *blockStore) DeleteSnapshot(snapshotID string) error {
 }
 
 func getFullDiskName(subscription string, resourceGroup string, diskName string) string {
-	return fmt.Sprintf("/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Compute/disks/%v", subscription, resourceGroup, diskName)
+	return fmt.Sprintf(diskURIFormatString, subscription, resourceGroup, diskName)
 }
 
-func getFullSnapshotName(subscription string, resourceGroup string, snapshotName string) string {
-	return fmt.Sprintf("/subscriptions/%v/resourceGroups/%v/providers/Microsoft.Compute/snapshots/%v", subscription, resourceGroup, snapshotName)
+// parseFullSnapshotName takes a snapshot URI and returns the subscription ID,
+// resource group, and snapshot name, or an error if the URI does not match
+// the regexp.
+func parseFullSnapshotName(name string) (string, string, string, error) {
+	submatches := snapshotURIRegexp.FindStringSubmatch(name)
+	if len(submatches) != len(snapshotURIRegexp.SubexpNames()) {
+		return "", "", "", errors.New("snapshot URI could not be parsed")
+	}
+
+	return submatches[1], submatches[2], submatches[3], nil
```

Comment:
Might be overkill, but if you're doing named capture groups, it might be worth looking up the matches by name (https://stackoverflow.com/questions/20750843/using-named-matches-from-go-regex).